ci/authn-gcp/run_gcloud.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash -ex
+
+# Script that runs a Google SDK container and executes a script argument.
+
+# The name of the script to run in the Google SDK container.
+GCLOUD_SCRIPT=$1
+
+main() {
+  echo "run_gcloud.sh with script: $GCLOUD_SCRIPT"
+  validate_pre_requisites || exit 1
+  write_sa_key_to_file || exit 1
+  run_gcloud_container || exit 1
+  echo "-> run_gcloud.sh with script: $GCLOUD_SCRIPT done"
+}
+
+validate_pre_requisites() {
+  echo 'validate_pre_requisites'
+
+  if [ -z "$GCLOUD_SCRIPT" ]; then
+    echo "-> Missing script file name argument"
+    exit 1
+  fi
+
+  if [ -z "$GCP_OWNER_SERVICE_KEY" ]; then
+    echo "-- env var 'GCP_OWNER_SERVICE_KEY' is undefined"
+    exit 1
+  fi
+
+  if [ -z "$GCP_OWNER_SERVICE_KEY_FILE" ]; then
+    echo "-- env var 'GCP_OWNER_SERVICE_KEY_FILE' is undefined"
+    exit 1
+  fi
+
+  if [ -z "$GCP_FETCH_TOKEN_FUNCTION" ]; then
+    echo "-- env var 'GCP_FETCH_TOKEN_FUNCTION' is undefined"
+    exit 1
+  fi
+
+  if [ -z "$GCP_PROJECT" ]; then
+    echo "-- env var 'GCP_PROJECT' is undefined"
+    exit 1
+  fi
+
+  if [ -z "$IDENTITY_TOKEN_FILE" ]; then
+    echo "-- env var 'IDENTITY_TOKEN_FILE' is undefined"
+    exit 1
+  fi
+  echo '-> validate_pre_requisites done'
+}
+
+# Authentication with Google CLI requires the service account key as file on disk.
+# This function writes the key that gets provisioned by summon temporary to disk, the cleanup_function.sh script
+# will delete it in the post/always branch of the stage.
+# The file is used by both by the deploy and clean function scripts.
+write_sa_key_to_file() {
+  echo 'write_sa_key_to_file'
+  if [ ! -f "$GCP_OWNER_SERVICE_KEY" ]; then
+    echo "-- Write account service key to file: '$GCP_OWNER_SERVICE_KEY_FILE' (required for gcloud auth)"
```

Comment:
@eranha is it possible to write the token right before the `gcloud auth command` and delete it right after, since it is a strong credential which may be left on the Jenkins machine in case the cleanup phase will failed ?

@sashaCher WDYT? we are using here google service account key to authenticate GCP in order to deploy a google function for the integration tests

Reply:
The file is used from within the Google SDK container to deploy the function and deleted by the clean function script.

Reply:
I run the gcloud container in run_glcoud and there I write the file to disk. I moved the funciton that deletes to run_gcloud script after the container exists.

Reply:
since we are in a side branch this issue should not be a blocker, but i really want Sasha to review it as well. so maybe you can take it with him offline ?